### DISCUSSION TITLE
I'm using term-image for XMPP powered A/V Calls

### DISCUSSION THREAD

**Discussion by goffi-contrib** [AnonymouX47/term-image#105] [Show and tell]
Hello,

Thanks for your neat work on term-image. I'm using it to show video stream (peer video and local feedback) in the CLI frontend of [Libervia](https://libervia.org), the XMPP client I'm working on.

I've posted 2 demo videos on ActivityPub: https://mastodon.social/@Goffi/111942641269230029 .  The first one is using Kitty backend, and the other one the Block backend, both on Konsole in a call with Conversations on Android.

The control buttons are made with `rich`. I'll probably write a blog post about it at some point.

**Comment by AnonymouX47** [comment]
Hurray! 🥳 🥳 🥳 

I really appreciate the compliments and you telling me about your project, means a lot to me. 🙏🏾

I checked out your project website and the demo videos, they're really great! Well done. 🚀
Unfortunately I won't be able to test out your project at the moment due to my tight schedule. 😢

Also, I'd like to let you know that a lot is on the way that would be particularly beneficial to your use case. I'm currently reworking the API in order to expand the scope of the project and increase the flexibility of the API. Already, I can see things such as [`term_image.renderable.FrameCount.INDEFINITE`](https://term-image.readthedocs.io/en/latest/api/renderable.html#term_image.renderable.FrameCount.INDEFINITE) and [`term_image.render.RenderIterator`](https://term-image.readthedocs.io/en/latest/api/render.html#term_image.render.RenderIterator) (**NOTE: these are yet to be released**) being of great benefit to your use case, eliminating unnecessary and possibly inefficient workaround/hacks. Also, see the [roadmap](https://github.com/AnonymouX47/term-image/milestones) and #99.

Thanks so much and I wish you the very best with your project! 🤗
